kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/records/BatchAwareMemoryRecordsBuilder.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption.records;
+
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.record.CompressionType;
+import org.apache.kafka.common.record.EndTransactionMarker;
+import org.apache.kafka.common.record.MemoryRecords;
+import org.apache.kafka.common.record.MemoryRecordsBuilder;
+import org.apache.kafka.common.record.Record;
+import org.apache.kafka.common.record.RecordBatch;
+import org.apache.kafka.common.record.SimpleRecord;
+import org.apache.kafka.common.record.TimestampType;
+import org.apache.kafka.common.utils.ByteBufferOutputStream;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A builder of {@link MemoryRecords} that can cope with multiple batches.
+ * (Apache Kafka's own {@link MemoryRecordsBuilder} assumes a single batch).
+ */
+public class BatchAwareMemoryRecordsBuilder {
```

Comment:
I don't believe this class intends to be threadsafe.  
```suggestion
@NotThreadSafe
public class BatchAwareMemoryRecordsBuilder {
```